doc/source/tutorial/stats.rst

```diff
@@ -1190,3 +1191,246 @@ because the p-value is very low and the MGC test statistic is relatively high.
 The MGC-map indicates a **strongly nonlinear relationship**. The optimal scale
 in this case is **equivalent to the local scale**, marked by a red spot on the
 map.
+
+Quasi-Monte Carlo
+-----------------
+
+Before talking about Quasi-Monte Carlo (QMC), a quick introduction about Monte
+Carlo (MC). The MC method was called in statistics 'model sampling'. It is
+a method which aims at trying out hypothesis. Once all the hypothesis we
+have defined have been tested, we can compute some statistics such as the
+mean of a given quantity of interest. A scenarios is defined by some variables.
+So, in order to map all scenarios, we use a random number generator.
+
+Generating random numbers is a more complex problem than it sounds. Simple
+Monte Carlo (MC) methods are designed to sample points to be independent and
+identically distributed (IID). But generating multiple sets of random points
+can produce radically different results.
+
+.. plot:: tutorial/stats/plots/qmc_plot_mc.py
+   :align: center
+   :include-source: 0
+
+In both cases, any points is taken randomly without any knowledge about the
+previous draws. And it is clear that some regions of the space are left
+unexplored. But as the number of sample increases, the space will fill.
+
+A great benefit of MC is that it has some convergence properties.
+Let's look at the mean of the square of sum in 5 dimensions:
+
+.. math::
+
+    f(\mathbf{x}) = \left( \sum_{j=1}^{5}x_j \right)^2,
+
+which has a known mean :math:`\mu = 5/3+5(5-1)/4`. Using MC sampling, we can
+compute the mean and the error follows a theoretical rate
+of :math:`O(n^{-1/2})`.
+
+.. plot:: tutorial/stats/plots/qmc_plot_conv_mc.py
+   :align: center
+   :include-source: 0
+
+Although the convergence is ensured, practitioner tend to want to have an
+exploration process which is more deterministic.
+What is commonly done to walk through all the hypothesis is
+to use a grid, also called a saturated design. Let’s consider the
+unit-hypercube, all bounds range from 0 to 1.
+Now, having a distance of 0.1 between each point, the number of points
+required to fill the unit interval would be 10. In a 2-dimensional hypercube
+the same spacing would require 100 and in 3-dimensions 1 000 points. As the
+number of dimensions goes, the number of experiments which is required to fill
+the space evenly rises exponentially as the volume of the space increases.
+This exponential growth is called the curse-of-dimensionality.
+
+    >>> import numpy as np
+    >>> disc = 10
+    >>> x1 = np.linspace(0, 1, disc)
+    >>> x2 = np.linspace(0, 1, disc)
+    >>> x3 = np.linspace(0, 1, disc)
+    >>> x1, x2, x3 = np.meshgrid(x1, x2, x3)
+
+.. plot:: tutorial/stats/plots/qmc_plot_curse.py
+   :align: center
+   :include-source: 0
+
+To mitigate this issue, QMC methods have been designed.
+They are deterministic, have a good coverage of the space and
+some of them can be continued and retain good properties.
```

Comment:
Some because LHS for instance requires a fix number of sample. And you cannot add a new point without breaking the LHS. There are versions of LHS that you can extend, but quite complicated algorithms (I know a few examples in the literature, but never experiment. I never saw this anywhere implemented).